pkg/runtime/config.go

```diff
@@ -346,6 +348,18 @@ func (c *Config) toInternal() (*internalConfig, error) {
 		}
 	}
 
+	if c.ApplicationHealthPort != "" {
+		intc.appConnectionConfig.HealthPort, err = strconv.Atoi(c.ApplicationHealthPort)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing app-port: %w", err)
+		}
+	} else if c.ApplicationPort != "" {
+		intc.appConnectionConfig.HealthPort, err = strconv.Atoi(c.ApplicationPort)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing app-port: %w", err)
+		}
+	}
+
```

Comment:
You need to check whether the app health port conflicts with the `intc.httpPort` and `intc.apiGRPPort` below as well.